tests/test_pl.py

```diff
@@ -140,3 +146,73 @@ def test_find_labels():
     f = "!A & (!AB & !A0)"
     formula = parser(f)
     assert formula.find_labels() == {c for c in {"A", "AB", "A0"}}
+
+
+def test_precedence():
```

Comment:
This could be translated into a "test class", with `setup_class` and `teardown_class` methods.

Some references:

https://docs.pytest.org/en/latest/getting-started.html#group-multiple-tests-in-a-class
https://docs.pytest.org/en/latest/xunit_setup.html#class-level-setup-teardown

Reply:
A problem with long test functions is that if at some point one of the assertions fails, the test is blocked.
Instead, splitting the assertions in several test function (eventually grouped in Test classes as suggested above) will not stop the test list if one assertion fails.

Reply:
However, this is just a general suggestion, feel free to ignore that.